csrc/ops/arith.cpp

```diff
@@ -2268,4 +2268,14 @@ TensorView* tensor(Val* val) {
   return out;
 }
 
+TensorView* argsort(
+    TensorView* inp,
+    int64_t dim,
+    bool descending,
+    bool stable) {
+  Val* out = ops::newValLike(inp, DataType::Int);
+  IrBuilder::create<ArgsortOp>(out, inp, dim, descending, stable);
```

Comment:
@naoyam here we are naively constructing an argsort operation.

I think the input dtype promotion you are referring to should happen here if applicable. If you look at the binary ops above, that's where the logic resides. https://github.com/NVIDIA/Fuser/blob/43c9e1310b50171e6dcbbb7e2769578a870ef9dd/csrc/ops/arith.cpp#L883-L885